packages/lesswrong/server/dialogues/cron.ts

```diff
@@ -21,3 +24,81 @@ addCronJob({
     })
   }
 });
+
+const checkCompatible = (preferences:DbDialogueMatchPreference[]) => {
+  if (preferences.length >= 2) {
+
+    // Topic
+    const sourceUserYesTopics = getUserTopics(preferences[0]);
+    const targetUserYesTopics = getUserTopics(preferences[1]);
+    const sharedTopics = sourceUserYesTopics.filter(topic => targetUserYesTopics.includes(topic));
+
+    // Format 
+    const { sync: userSync, async: userAsync } = getSyncAsyncPreferences(preferences[0]);
+    const { sync: targetUserSync, async: targetUserAsync } = getSyncAsyncPreferences(preferences[1]);
+    const { formatPreferenceMatch } = getPreferenceMatches(
+      { sync: userSync, async: userAsync },
+      { sync: targetUserSync, async: targetUserAsync }
+    );
+
+    if (sharedTopics.length > 0 && formatPreferenceMatch) {
```

Comment:
Prefer to do an early return if `sharedTopics.length === 0` rather than running through the rest of the logic (even though it won't matter in that case).